eng/docgeneration/Generate-DocIndex.ps1

```diff
@@ -12,153 +27,95 @@ Write-Verbose "Initializing Default DocFx Site..."
-            # handled
-            if (("" -eq $parent) -or ($null -eq $parent))
-            {
-                # Cosmos has a root pom which includes pom.client.xml that won't
-                # be detected by this logic. It's easier to deal with specially
-                # than it is to try and climb the pom chain here.
-                if ($Dir.BaseName -eq 'cosmos') 
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    Write-Host "*snowflake* Pom $PomPath, has a null or empty relative path."
-                }
-            } 
-            else 
-            {
-                if (($parent.IndexOf('azure-client-sdk-parent') -ne -1) -or ($parent.IndexOf('azure-data-sdk-parent') -ne -1))
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    $mgmtArr += $artifactId
-                }
-            }
-        }
+    else {
+        # Next page call
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex&marker=$pageToken"
     }
-    # Only create this if there's something to create
-    #if (($clientArr.Count -gt 0) -or ($mgmtArr.Count -gt 0))
-    if ($clientArr.Count -gt 0)
-    {
-    New-Item -Path $YmlPath -Name "$($Dir.Name).md" -Force
-    Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: $($mappedDir)`r`n  href: $($Dir.Name).md"
-    # loop through the arrays and add the appropriate artifacts under the appropriate headings
-    if ($clientArr.Count -gt 0)
-    {
-        Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "# Client Libraries"
-        foreach($lib in $clientArr) 
-        {
-            Write-Host "Write $($lib) to $($Dir.Name).md"
-            Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "#### $lib"
+    # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+    $rawConent = $resp.Substring(3)
+    # Convert to xml documents.
+    $xmlDoc = [xml]$rawConent
+    foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+        # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+        $aritifact = $elem.Name -replace $regex, '$1'
+        # Some languages need to convert the artifact name, e.g azure-data-appconfiguration -> @azure/data-appconfiguration
+        if ($packageNameRegex) {
+            $aritifact = $aritifact -replace $packageNameRegex, $packageNameReplacement
+        }
+        # Read the artifact package infomation from csv of Azure/azure-sdk/_data/release/latest repo.
+        $packageInfo = $metadata | ? { $_.Package -Contains $aritifact}
+        if (!$packageInfo) {
+            Write-Error "Did not find the artifacts from release csv. Please check and update."
```

Comment:
We should be careful about failing here, especially for new packages I expect this will fail for a little while. Instead we should just use our fallback like below where you don't find a ServiceName.